activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -74,8 +74,8 @@ def initialize(base, table, associations, join_type)
         @join_type = join_type
       end
 
-      def base_klass
-        join_root.base_klass
+      def base_class
```

Comment:
This is a publicly available API. Deprecate?

Reply:
It is not. See the class definition. There is a `:nodoc:`